cmake/HalideGeneratorHelpers.cmake

```diff
@@ -244,16 +244,23 @@ function(add_halide_library TARGET)
         if (NOT TARGET Halide::Python)
             message(FATAL_ERROR "This version of Halide was built without support for Python bindings; rebuild using WITH_PYTHON_BINDINGS=ON to use this rule with Python Generators.")
         endif ()
+        
         if (NOT TARGET Python3::Interpreter)
             message(FATAL_ERROR "You must call find_package(Python3) in your CMake code in order to use this rule with Python Generators.")
         endif ()
-        set(PYTHONPATH "$<TARGET_FILE_DIR:Halide::Python>/..")
-        if (Halide_SHARED_ASAN_RUNTIME_LIBRARY)
-            # ASAN's Leak detector will report that we leaked all the PyBind11
-            # Module-support code, so disable it here.
-            set(EXTRA_ENV_VARS ASAN_OPTIONS=detect_leaks=0 ${Halide_SANITIZER_ENV_VARS})
+
+        if (CMAKE_VERSION VERSION_LESS 3.24)
```

Comment:
...do I even want to know what this is about?

Reply:
The `--` is a pretty common feature that indicates "end of flags". The remaining arguments are not parsed for flags. It's basically a macro hygiene thing. CMake only started supporting it as of 3.24, though.